# National Health and Nutrition Examination Survey

https://www.cdc.gov/nchs/nhanes/index.htm

Can I predict diabetes based off of diet?

Be able to predict diabetes, prediabetes, and not diabetic based on:
* Lab Data
    * Insulin
    * Glucose
    * HDL
    * Total Cholesterol
* Examination Data
    * Body Measurements
    * Blood Preasure
* Diet Interview
    * 1st and 2nd Day Nutrition
* Demographics
* Questionnaire Data
    * Alcohol
    * Smoking

Look into finding the diet of people who have diabetes?

***

# Pandas Tools

#### Count up nans in a column
* df['Column_Name'].isnull().sum(axis = 0)

#### Fill all NaN values in a column with a particular value
* Demographics_df['DMDCITZN'].fillna(2.0, inplace=True)

##### Combining Dataframes

* Full_Days_Nutrition_df = pd.merge(Day_1_Nutrition_df, Day_2_Nutrition_df, on="SEQN")
* Full_Days_Nutrition_df.head()

##### Find rows that are all NaN

* nan = df[df.isnull().all(axis=1)].index
* nan

##### Change all '99' in a column to 1
Demographics_df.loc[Demographics_df['DMDBORN4'] == 99, 'DMDBORN4'] = 1

##### Check for Nans

* df.isnull().values.any()

##### Getting Dummies
* Finished_Demographics_df = pd.get_dummies(Demographics_df, columns = need_dummies, drop_first = True)

##### Rename Columns
* df = df.rename(columns={'oldName1': 'newName1', 'oldName2': 'newName2'})

##### Add 'Age', then drop all 0-17 year olds
* Blood_Pressure_df = pd.merge(Blood_Pressure_df, Age_df, on="SEQN")
* Blood_Pressure_df = Blood_Pressure_df.drop(Blood_Pressure_df[Blood_Pressure_df.Age < 18].index)
* Blood_Pressure_df.head()

***

# Import Libraries

In [91]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt


import warnings
warnings.filterwarnings("ignore")


***

<div class="alert alert-block alert-info">

# Diet Interview Data - Done, Done, Done, Done!!!!!

# Final DF - Average_Days_Nutrition_df   -  (5735, 68)
# Still missing 469 complete rows!!!!
### Next Steps:
* Figure out how to combine Day 1 and Day 2 Nutrition
    * If a row is missing one of the days
        * fill in with the other day's info
    * Should I combine the 2 days or keep them seperate?
        * If I do:
            * Sum or Average?
                * Verdict - Average
    
* After combining and filling in NaNs from the other day, I'm still missing 1205 rows
    * Some of these may be dropped after I remove ages 0-17




* Not using Individual Foods (Day 1 and 2)
    * It was pretty much the same data as the Nutrition dataframes


### Load the data

In [336]:
Preprocess_Day_1_Nutrition_df = pd.read_sas(filepath_or_buffer = 'Diet_Interview_Total_Nutrient_Intakes_First_Day.XPT')
Preprocess_Day_2_Nutrition_df = pd.read_sas(filepath_or_buffer = 'Diet_Interview_Total_Nutrient_Intakes_Second_Day.XPT')

# Preprocess_IF1_df = pd.read_sas(filepath_or_buffer = 'Diet_Interview_Individual_Foods_First_Day.XPT')
# Preprocess_IF2_df = pd.read_sas(filepath_or_buffer = 'Diet_Interview_Individual_Foods_Second_Day.XPT')
# Diet_Interview_Food_Codes = pd.read_sas(filepath_or_buffer = 'Diet_Inteview_Food_Codes.XPT')


#### Day 1 Nutrition

   * Missing 1217
   * DR1TKCAL - Energy (kcal)
   * DR1TPROT - Protein (gm)
   * DR1TCARB - Carbohydrate (gm)
   * DR1TSUGR - Total sugars (gm)
   * DR1TFIBE - Dietary fiber (gm)
   * DR1TTFAT - Total fat (gm)
   * DR1TSFAT - Total saturated fatty acids (gm)
   * DR1TMFAT - Total monounsaturated fatty acids (gm)
   * DR1TPFAT - Total polyunsaturated fatty acids (gm)
   * DR1TCHOL - Cholesterol (mg)
   * DR1TATOC - Vitamin E as alpha-tocopherol (mg)
   * DR1TATOA - Added alpha-tocopherol (Vitamin E) (mg)
   * DR1TRET - Retinol (mcg)
   * DR1TVARA - Vitamin A, RAE (mcg)
   * DR1TACAR - Alpha-carotene (mcg)
   * DR1TBCAR - Beta-carotene (mcg)
   * DR1TCRYP - Beta-cryptoxanthin (mcg)
   * DR1TLYCO - Lycopene (mcg)
   * DR1TLZ - Lutein + zeaxanthin (mcg)
   * DR1TVB1 - Thiamin (Vitamin B1) (mg)
   * DR1TVB2 - Riboflavin (Vitamin B2) (mg)
   * DR1TNIAC - Niacin (mg)
   * DR1TVB6 - Vitamin B6 (mg)
   * DR1TFOLA - Total folate (mcg)
   * DR1TFA - Folic acid (mcg)
   * DR1TFF - Food folate (mcg)
   * DR1TFDFE - Folate, DFE (mcg)
   * DR1TCHL - Total choline (mg)
   * DR1TVB12 - Vitamin B12 (mcg)
   * DR1TB12A - Added vitamin B12 (mcg)
   * DR1TVC - Vitamin C (mg)
   * DR1TVD - Vitamin D (D2 + D3) (mcg)
   * DR1TVK - Vitamin K (mcg)
   * DR1TCALC - Calcium (mg)
   * DR1TPHOS - Phosphorus (mg)
   * DR1TMAGN - Magnesium (mg)
   * DR1TIRON - Iron (mg)
   * DR1TZINC - Zinc (mg)
   * DR1TCOPP - Copper (mg)
   * DR1TSODI - Sodium (mg)
   * DR1TPOTA - Potassium (mg)
   * DR1TSELE - Selenium (mcg)
   * DR1TCAFF - Caffeine (mg)
   * DR1TTHEO - Theobromine (mg)
   * DR1TALCO - Alcohol (gm)
   * DR1TMOIS - Moisture (gm)
   * DR1TS040 - SFA 4:0 (Butanoic) (gm)
   * DR1TS060 - SFA 6:0 (Hexanoic) (gm)
   * DR1TS080 - SFA 8:0 (Octanoic) (gm)
   * DR1TS100 - SFA 10:0 (Decanoic) (gm)
   * DR1TS120 - SFA 12:0 (Dodecanoic) (gm)
   * DR1TS140 - SFA 14:0 (Tetradecanoic) (gm)
   * DR1TS160 - SFA 16:0 (Hexadecanoic) (gm)
   * DR1TS180 - SFA 18:0 (Octadecanoic) (gm)
   * DR1TM161 - MFA 16:1 (Hexadecenoic) (gm)
   * DR1TM181 - MFA 18:1 (Octadecenoic) (gm)
   * DR1TM201 - MFA 20:1 (Eicosenoic) (gm)
   * DR1TM221 - MFA 22:1 (Docosenoic) (gm)
   * DR1TP182 - PFA 18:2 (Octadecadienoic) (gm)
   * DR1TP183 - PFA 18:3 (Octadecatrienoic) (gm)
   * DR1TP184 - PFA 18:4 (Octadecatetraenoic) (gm)
   * DR1TP204 - PFA 20:4 (Eicosatetraenoic) (gm)
   * DR1TP205 - PFA 20:5 (Eicosapentaenoic) (gm)
   * DR1TP225 - PFA 22:5 (Docosapentaenoic) (gm)
   * DR1TP226 - PFA 22:6 (Docosahexaenoic) (gm)
   * DR1_320Z - Total plain water drank yesterday (gm)
   
   
   * Missing 1038  

In [337]:
day_1_nutrient_columns = ['SEQN','DR1TKCAL', 'DR1TPROT', 'DR1TCARB', 'DR1TSUGR', 'DR1TFIBE', 'DR1TTFAT', 'DR1TSFAT',
                          'DR1TMFAT', 'DR1TPFAT', 'DR1TCHOL', 'DR1TATOC', 'DR1TATOA', 'DR1TRET', 'DR1TVARA', 'DR1TACAR',
                          'DR1TBCAR', 'DR1TCRYP', 'DR1TLYCO', 'DR1TLZ', 'DR1TVB1', 'DR1TVB2', 'DR1TNIAC', 'DR1TVB6',
                          'DR1TFOLA', 'DR1TFA', 'DR1TFF', 'DR1TFDFE', 'DR1TCHL', 'DR1TVB12', 'DR1TB12A', 'DR1TVC',
                          'DR1TVD', 'DR1TVK', 'DR1TCALC', 'DR1TPHOS', 'DR1TMAGN', 'DR1TIRON', 'DR1TZINC', 'DR1TCOPP',
                          'DR1TSODI', 'DR1TPOTA', 'DR1TSELE', 'DR1TCAFF', 'DR1TTHEO', 'DR1TALCO', 'DR1TMOIS',
                          'DR1TS040', 'DR1TS060', 'DR1TS080', 'DR1TS100', 'DR1TS120', 'DR1TS140', 'DR1TS160', 
                          'DR1TS180', 'DR1TM161', 'DR1TM181', 'DR1TM201', 'DR1TM221', 'DR1TP182', 'DR1TP183',
                          'DR1TP184', 'DR1TP204', 'DR1TP205', 'DR1TP225', 'DR1TP226', 'DR1_320Z']

In [338]:
Day_1_Nutrition_df = Preprocess_Day_1_Nutrition_df[day_1_nutrient_columns]
Day_1_Nutrition_df.head()

SEQN  DR1TKCAL  DR1TPROT  DR1TCARB  DR1TSUGR  DR1TFIBE  DR1TTFAT  \
0  83732.0    1781.0     76.03    193.29     42.31      23.6     79.24   
1  83733.0    2964.0     62.36    356.85    180.84       7.3     77.91   
2  83734.0    2482.0    116.50    304.53     62.87      35.1     91.97   
3  83735.0    1340.0     70.38    157.99     54.77      14.0     49.23   
4  83736.0     604.0     19.30     90.30     71.84       2.0     19.63   

   DR1TSFAT  DR1TMFAT  DR1TPFAT      ...       DR1TM201      DR1TM221  \
0    23.430    31.897    18.528      ...          0.332  2.760000e-01   
1    25.722    19.098    19.216      ...          0.155  5.397605e-79   
2    23.122    28.666    32.561      ...          0.420  2.200000e-02   
3    17.593    18.327     9.588      ...          0.258  3.600000e-02   
4     4.581     8.504     3.889      ...          0.101  1.200000e-02   

   DR1TP182  DR1TP183      DR1TP184  DR1TP204  DR1TP205  DR1TP225  DR1TP226  \
0    16.024     2.294  4.000000e-03     0.110     0.004     0.017     0.002   
1    16.712     2.140  5.397605e-79     0.206     0.005     0.026     0.036   
2    28.405     3.365  5.397605e-79     0.360     0.007     0.026     0.062   
3     8.370     1.001  4.000000e-03     0.076     0.015     0.017     0.019   
4     3.378     0.430  1.000000e-03     0.038     0.001     0.002     0.001   

       DR1_320Z  
0  9.600000e+02  
1  5.070000e+02  
2  8.700000e+02  
3  1.800000e+02  
4  5.397605e-79  

[5 rows x 67 columns]

In [339]:
Day_2_Nutrition_df['DR2TPROT'].isnull().sum(axis = 0)

2669

#### Day 2 Nutrition

* Missing 2669
* DR2TKCAL - Energy (kcal)
* DR2TPROT - Protein (gm)
* DR2TCARB - Carbohydrate (gm)
* DR2TSUGR - Total sugars (gm)
* DR2TFIBE - Dietary fiber (gm)
* DR2TTFAT - Total fat (gm)
* DR2TSFAT - Total saturated fatty acids (gm)
* DR2TMFAT - Total monounsaturated fatty acids (gm)
* DR2TPFAT - Total polyunsaturated fatty acids (gm)
* DR2TCHOL - Cholesterol (mg)
* DR2TATOC - Vitamin E as alpha-tocopherol (mg)
* DR2TATOA - Added alpha-tocopherol (Vitamin E) (mg)
* DR2TRET - Retinol (mcg)
* DR2TVARA - Vitamin A, RAE (mcg)
* DR2TACAR - Alpha-carotene (mcg)
* DR2TBCAR - Beta-carotene (mcg)
* DR2TCRYP - Beta-cryptoxanthin (mcg)
* DR2TLYCO - Lycopene (mcg)
* DR2TLZ - Lutein + zeaxanthin (mcg)
* DR2TVB1 - Thiamin (Vitamin B1) (mg)
* DR2TVB2 - Riboflavin (Vitamin B2) (mg)
* DR2TNIAC - Niacin (mg)
* DR2TVB6 - Vitamin B6 (mg)
* DR2TFOLA - Total folate (mcg)
* DR2TFA - Folic acid (mcg)
* DR2TFF - Food folate (mcg)
* DR2TFDFE - Folate, DFE (mcg)
* DR2TCHL - Total choline (mg)
* DR2TVB12 - Vitamin B12 (mcg)
* DR2TB12A - Added vitamin B12 (mcg)
* DR2TVC - Vitamin C (mg)
* DR2TVD - Vitamin D (D2 + D3) (mcg)
* DR2TVK - Vitamin K (mcg)
* DR2TCALC - Calcium (mg)
* DR2TPHOS - Phosphorus (mg)
* DR2TMAGN - Magnesium (mg)
* DR2TIRON - Iron (mg)
* DR2TZINC - Zinc (mg)
* DR2TCOPP - Copper (mg)
* DR2TSODI - Sodium (mg)
* DR2TPOTA - Potassium (mg)
* DR2TSELE - Selenium (mcg)
* DR2TCAFF - Caffeine (mg)
* DR2TTHEO - Theobromine (mg)
* DR2TALCO - Alcohol (gm)
* DR2TMOIS - Moisture (gm)
* DR2TS040 - SFA 4:0 (Butanoic) (gm)
* DR2TS060 - SFA 6:0 (Hexanoic) (gm)
* DR2TS080 - SFA 8:0 (Octanoic) (gm)
* DR2TS100 - SFA 10:0 (Decanoic) (gm)
* DR2TS120 - SFA 12:0 (Dodecanoic) (gm)
* DR2TS140 - SFA 14:0 (Tetradecanoic) (gm)
* DR2TS160 - SFA 16:0 (Hexadecanoic) (gm)
* DR2TS180 - SFA 18:0 (Octadecanoic) (gm)
* DR2TM161 - MFA 16:1 (Hexadecenoic) (gm)
* DR2TM181 - MFA 18:1 (Octadecenoic) (gm)
* DR2TM201 - MFA 20:1 (Eicosenoic) (gm)
* DR2TM221 - MFA 22:1 (Docosenoic) (gm)
* DR2TP182 - PFA 18:2 (Octadecadienoic) (gm)
* DR2TP183 - PFA 18:3 (Octadecatrienoic) (gm)
* DR2TP184 - PFA 18:4 (Octadecatetraenoic) (gm)
* DR2TP204 - PFA 20:4 (Eicosatetraenoic) (gm)
* DR2TP205 - PFA 20:5 (Eicosapentaenoic) (gm)
* DR2TP225 - PFA 22:5 (Docosapentaenoic) (gm)
* DR2TP226 - PFA 22:6 (Docosahexaenoic) (gm)
* DR2_320Z - Total plain water drank yesterday (gm)

In [340]:
day_2_nutrient_columns = ['SEQN', 'DR2TKCAL', 'DR2TPROT', 'DR2TCARB', 'DR2TSUGR', 'DR2TFIBE', 'DR2TTFAT', 'DR2TSFAT',
                          'DR2TMFAT', 'DR2TPFAT', 'DR2TCHOL', 'DR2TATOC', 'DR2TATOA', 'DR2TRET', 'DR2TVARA', 'DR2TACAR',
                          'DR2TBCAR', 'DR2TCRYP', 'DR2TLYCO', 'DR2TLZ', 'DR2TVB1', 'DR2TVB2', 'DR2TNIAC', 'DR2TVB6',
                          'DR2TFOLA', 'DR2TFA', 'DR2TFF', 'DR2TFDFE', 'DR2TCHL', 'DR2TVB12', 'DR2TB12A', 'DR2TVC',
                          'DR2TVD', 'DR2TVK', 'DR2TCALC', 'DR2TPHOS', 'DR2TMAGN', 'DR2TIRON', 'DR2TZINC', 'DR2TCOPP',
                          'DR2TSODI', 'DR2TPOTA', 'DR2TSELE', 'DR2TCAFF', 'DR2TTHEO', 'DR2TALCO', 'DR2TMOIS',
                          'DR2TS040', 'DR2TS060', 'DR2TS080', 'DR2TS100', 'DR2TS120', 'DR2TS140', 'DR2TS160', 
                          'DR2TS180', 'DR2TM161', 'DR2TM181', 'DR2TM201', 'DR2TM221', 'DR2TP182', 'DR2TP183',
                          'DR2TP184', 'DR2TP204', 'DR2TP205', 'DR2TP225', 'DR2TP226', 'DR2_320Z']

In [341]:
Day_2_Nutrition_df = Preprocess_Day_2_Nutrition_df[day_2_nutrient_columns]
Day_2_Nutrition_df.head()

SEQN  DR2TKCAL  DR2TPROT  DR2TCARB  DR2TSUGR  DR2TFIBE  DR2TTFAT  \
0  83732.0    2744.0    133.60    281.64    118.40      15.9    121.59   
1  83733.0    3971.0    165.86    223.57     34.52      12.2    154.55   
2  83734.0    1928.0     93.78    194.17     94.51      17.4     90.14   
3  83735.0       NaN       NaN       NaN       NaN       NaN       NaN   
4  83736.0    1102.0     31.29    131.83     73.63       3.1     50.88   

   DR2TSFAT  DR2TMFAT  DR2TPFAT      ...       DR2TM201  DR2TM221  DR2TP182  \
0    40.420    42.428    26.813      ...          0.486     0.112    23.147   
1    32.969    61.039    47.020      ...          0.924     0.027    39.908   
2    39.829    30.152    10.465      ...          0.215     0.017     9.087   
3       NaN       NaN       NaN      ...            NaN       NaN       NaN   
4    21.936    17.853     5.942      ...          0.194     0.013     5.194   

   DR2TP183      DR2TP184  DR2TP204  DR2TP205  DR2TP225  DR2TP226  \
0     2.927  5.397605e-79     0.350     0.013     0.025     0.061   
1     4.940  1.900000e-02     0.498     0.136     0.117     0.489   
2     1.016  6.300000e-02     0.151     0.009     0.043     0.001   
3       NaN           NaN       NaN       NaN       NaN       NaN   
4     0.534  1.000000e-03     0.078     0.002     0.008     0.003   

       DR2_320Z  
0  1.440000e+03  
1  4.800000e+02  
2  7.200000e+02  
3           NaN  
4  5.397605e-79  

[5 rows x 67 columns]

# Cleaning Nutrition Data

In [342]:
Full_Days_Nutrition_df = pd.merge(Day_1_Nutrition_df, Day_2_Nutrition_df, on="SEQN")
Full_Days_Nutrition_df.head()

SEQN  DR1TKCAL  DR1TPROT  DR1TCARB  DR1TSUGR  DR1TFIBE  DR1TTFAT  \
0  83732.0    1781.0     76.03    193.29     42.31      23.6     79.24   
1  83733.0    2964.0     62.36    356.85    180.84       7.3     77.91   
2  83734.0    2482.0    116.50    304.53     62.87      35.1     91.97   
3  83735.0    1340.0     70.38    157.99     54.77      14.0     49.23   
4  83736.0     604.0     19.30     90.30     71.84       2.0     19.63   

   DR1TSFAT  DR1TMFAT  DR1TPFAT      ...       DR2TM201  DR2TM221  DR2TP182  \
0    23.430    31.897    18.528      ...          0.486     0.112    23.147   
1    25.722    19.098    19.216      ...          0.924     0.027    39.908   
2    23.122    28.666    32.561      ...          0.215     0.017     9.087   
3    17.593    18.327     9.588      ...            NaN       NaN       NaN   
4     4.581     8.504     3.889      ...          0.194     0.013     5.194   

   DR2TP183      DR2TP184  DR2TP204  DR2TP205  DR2TP225  DR2TP226  \
0     2.927  5.397605e-79     0.350     0.013     0.025     0.061   
1     4.940  1.900000e-02     0.498     0.136     0.117     0.489   
2     1.016  6.300000e-02     0.151     0.009     0.043     0.001   
3       NaN           NaN       NaN       NaN       NaN       NaN   
4     0.534  1.000000e-03     0.078     0.002     0.008     0.003   

       DR2_320Z  
0  1.440000e+03  
1  4.800000e+02  
2  7.200000e+02  
3           NaN  
4  5.397605e-79  

[5 rows x 133 columns]

In [345]:
# Fill in missing Day 2 values with Day 1's 

for i in day_2_nutrient_columns[1:]:
    lst_i = list(i)
    lst_i[2] = '1'
    j = ''.join(lst_i)
    Full_Days_Nutrition_df[i] = Full_Days_Nutrition_df[i].fillna(value = Full_Days_Nutrition_df[j])

In [346]:
Full_Days_Nutrition_df.head()

SEQN  DR1TKCAL  DR1TPROT  DR1TCARB  DR1TSUGR  DR1TFIBE  DR1TTFAT  \
0  83732.0    1781.0     76.03    193.29     42.31      23.6     79.24   
1  83733.0    2964.0     62.36    356.85    180.84       7.3     77.91   
2  83734.0    2482.0    116.50    304.53     62.87      35.1     91.97   
3  83735.0    1340.0     70.38    157.99     54.77      14.0     49.23   
4  83736.0     604.0     19.30     90.30     71.84       2.0     19.63   

   DR1TSFAT  DR1TMFAT  DR1TPFAT      ...       DR2TM201  DR2TM221  DR2TP182  \
0    23.430    31.897    18.528      ...          0.486     0.112    23.147   
1    25.722    19.098    19.216      ...          0.924     0.027    39.908   
2    23.122    28.666    32.561      ...          0.215     0.017     9.087   
3    17.593    18.327     9.588      ...          0.258     0.036     8.370   
4     4.581     8.504     3.889      ...          0.194     0.013     5.194   

   DR2TP183      DR2TP184  DR2TP204  DR2TP205  DR2TP225  DR2TP226  \
0     2.927  5.397605e-79     0.350     0.013     0.025     0.061   
1     4.940  1.900000e-02     0.498     0.136     0.117     0.489   
2     1.016  6.300000e-02     0.151     0.009     0.043     0.001   
3     1.001  4.000000e-03     0.076     0.015     0.017     0.019   
4     0.534  1.000000e-03     0.078     0.002     0.008     0.003   

       DR2_320Z  
0  1.440000e+03  
1  4.800000e+02  
2  7.200000e+02  
3  1.800000e+02  
4  5.397605e-79  

[5 rows x 133 columns]

In [347]:
# Fill in missing Day 1 values with Day 2's 

for i in day_1_nutrient_columns[1:]:
    lst_i = list(i)
    lst_i[2] = '2'
    j = ''.join(lst_i)
    Full_Days_Nutrition_df[i] = Full_Days_Nutrition_df[i].fillna(value = Full_Days_Nutrition_df[j])

In [348]:
Full_Days_Nutrition_df.head()

SEQN  DR1TKCAL  DR1TPROT  DR1TCARB  DR1TSUGR  DR1TFIBE  DR1TTFAT  \
0  83732.0    1781.0     76.03    193.29     42.31      23.6     79.24   
1  83733.0    2964.0     62.36    356.85    180.84       7.3     77.91   
2  83734.0    2482.0    116.50    304.53     62.87      35.1     91.97   
3  83735.0    1340.0     70.38    157.99     54.77      14.0     49.23   
4  83736.0     604.0     19.30     90.30     71.84       2.0     19.63   

   DR1TSFAT  DR1TMFAT  DR1TPFAT      ...       DR2TM201  DR2TM221  DR2TP182  \
0    23.430    31.897    18.528      ...          0.486     0.112    23.147   
1    25.722    19.098    19.216      ...          0.924     0.027    39.908   
2    23.122    28.666    32.561      ...          0.215     0.017     9.087   
3    17.593    18.327     9.588      ...          0.258     0.036     8.370   
4     4.581     8.504     3.889      ...          0.194     0.013     5.194   

   DR2TP183      DR2TP184  DR2TP204  DR2TP205  DR2TP225  DR2TP226  \
0     2.927  5.397605e-79     0.350     0.013     0.025     0.061   
1     4.940  1.900000e-02     0.498     0.136     0.117     0.489   
2     1.016  6.300000e-02     0.151     0.009     0.043     0.001   
3     1.001  4.000000e-03     0.076     0.015     0.017     0.019   
4     0.534  1.000000e-03     0.078     0.002     0.008     0.003   

       DR2_320Z  
0  1.440000e+03  
1  4.800000e+02  
2  7.200000e+02  
3  1.800000e+02  
4  5.397605e-79  

[5 rows x 133 columns]

In [349]:
# get a list of every column title in the dataframe
full_day_columns = list(Full_Days_Nutrition_df.columns)

In [350]:
# fill all NaN with -1
for i in full_day_columns:
    Full_Days_Nutrition_df[i] = Full_Days_Nutrition_df[i].fillna(value = -1)

In [351]:
# create new columns with the average of the two days

for i in day_1_nutrient_columns[1:]:
    day_1 = i
    lst_i = list(i)
    
    lst_i[2] = '2'
    day_2 = ''.join(lst_i)
    
    lst_i[2] = '3'
    day_3 = ''.join(lst_i)
    
    Full_Days_Nutrition_df[day_3] = (Full_Days_Nutrition_df[day_1] + Full_Days_Nutrition_df[day_2]) / 2



In [352]:
# create a new df with just the average columns

Average_Days_Nutrition_df = Full_Days_Nutrition_df.drop(full_day_columns[1:], axis = 1)

In [353]:
Average_Days_Nutrition_df.head()

SEQN  DR3TKCAL  DR3TPROT  DR3TCARB  DR3TSUGR  DR3TFIBE  DR3TTFAT  \
0  83732.0    2262.5   104.815   237.465    80.355     19.75   100.415   
1  83733.0    3467.5   114.110   290.210   107.680      9.75   116.230   
2  83734.0    2205.0   105.140   249.350    78.690     26.25    91.055   
3  83735.0    1340.0    70.380   157.990    54.770     14.00    49.230   
4  83736.0     853.0    25.295   111.065    72.735      2.55    35.255   

   DR3TSFAT  DR3TMFAT  DR3TPFAT      ...       DR3TM201  DR3TM221  DR3TP182  \
0   31.9250   37.1625   22.6705      ...         0.4090    0.1940   19.5855   
1   29.3455   40.0685   33.1180      ...         0.5395    0.0135   28.3100   
2   31.4755   29.4090   21.5130      ...         0.3175    0.0195   18.7460   
3   17.5930   18.3270    9.5880      ...         0.2580    0.0360    8.3700   
4   13.2585   13.1785    4.9155      ...         0.1475    0.0125    4.2860   

   DR3TP183  DR3TP184  DR3TP204  DR3TP205  DR3TP225  DR3TP226      DR3_320Z  
0    2.6105    0.0020    0.2300    0.0085    0.0210    0.0315  1.200000e+03  
1    3.5400    0.0095    0.3520    0.0705    0.0715    0.2625  4.935000e+02  
2    2.1905    0.0315    0.2555    0.0080    0.0345    0.0315  7.950000e+02  
3    1.0010    0.0040    0.0760    0.0150    0.0170    0.0190  1.800000e+02  
4    0.4820    0.0010    0.0580    0.0015    0.0050    0.0020  5.397605e-79  

[5 rows x 67 columns]

In [354]:
# drop younger than 18
# Add age column first, then drop everyone yonger than 18
Average_Days_Nutrition_df = pd.merge(Average_Days_Nutrition_df, Age_df, on="SEQN")

In [355]:
Average_Days_Nutrition_df = Average_Days_Nutrition_df.drop(Average_Days_Nutrition_df[Average_Days_Nutrition_df.Age < 18].index)
Average_Days_Nutrition_df.shape

(5735, 68)

In [419]:
Average_Days_Nutrition_df.to_csv('Average_Days_Nutrition.csv')

***

<div class="alert alert-block alert-info">

# Demographics Data - Done, Done, Done, Done!!!!!

## Final DF - Finished_Demographics_df   -   (5992, 58)

## This dataframe has age!!!!!!

</div>

### Load the data

In [272]:
Preprocess_Demographics_df = pd.read_sas(filepath_or_buffer = 'Demographic_Variables_and_Sample_Weights.XPT')


##### Demographics Columns

* RIAGENDR - Gender (1=Male, 2=Female)
    * Male = 1
    * Female = 0
* RIDAGEYR - Age in years at screening
* RIDRETH3 - Race/Hispanic origin w/ NH Asian
    * Dummies (1,2,3,4,6,7)
* DMQMILIZ - Served active duty in US Armed Forces
    * Yes = 1
    * No = 2
    * NaN -> 2 (mode)
* DMDBORN4 - Born in the US
    * Yes = 1
    * No = 2
    * 99 -> 1 (mode)
* DMDCITZN - US Citizenship status (1=Citizen, 2=Not a Citizen)
    * Yes = 1
    * No = 2
    * 7 -> 2 (you're probably refusing becuse you're illegal)
    * 9, NaN -> 1 (mode)
* SIALANG - Language of Interview
    * English = 1
    * Spanish = 2
* DMDHHSIZ - Total number of people in the Household
    * Dummies (1-7)
* DMDFMSIZ - Total number of people in the Family
    * Dummies (1-7)
* DMDHHSZA - Total number of children 5 years or younger in HH
    * Dummies (0-3)
* DMDHHSZB - Total number of children 6-17 years old in HH
    * Dummies (0-4)
* DMDHHSZE - Total number of adults 60 years or older in HH
    * Dummies (0-3)
* DMDHRGND - Head of House ref persons gender
    * Male = 1
    * Female = 2
* DMDHRAGE - HH ref persons age in years
* DMDHREDU - HH ref persons education level
    * Dummies (1-5)
    * 9 -> 3
    * NaN -> 4
* DMDHRMAR - HH ref persons marital status
    * Dummies (1-6)
    * 77 -> 5 (never married)
    * 99 -> 6 (living with partner)
    * Nan -> 1 (mode)
* INDFMIN2 - Annual family income
    * Dummies (1-10, 12-15)
    * 77 -> 13 (under 20,000)
    * 99 -> 12 (20,000+ )
    * NaN -> 12 (20,000 +)

In [303]:
demo_column_list = ['SEQN','RIAGENDR', 'RIDAGEYR', 'RIDRETH3', 'DMQMILIZ', 'DMDBORN4', 'DMDCITZN', 'SIALANG', 'DMDHHSIZ', 'DMDFMSIZ','DMDHHSZA', 'DMDHHSZB', 'DMDHHSZE', 'DMDHRGND', 'DMDHRAGE', 'DMDHREDU','DMDHRMAR', 'INDFMIN2']
Demographics_df = Preprocess_Demographics_df[demo_column_list]
Demographics_df.head()


SEQN  RIAGENDR  RIDAGEYR  RIDRETH3  DMQMILIZ  DMDBORN4  DMDCITZN  \
0  83732.0       1.0      62.0       3.0       2.0       1.0       1.0   
1  83733.0       1.0      53.0       3.0       2.0       2.0       2.0   
2  83734.0       1.0      78.0       3.0       1.0       1.0       1.0   
3  83735.0       2.0      56.0       3.0       2.0       1.0       1.0   
4  83736.0       2.0      42.0       4.0       2.0       1.0       1.0   

   SIALANG  DMDHHSIZ  DMDFMSIZ      DMDHHSZA      DMDHHSZB      DMDHHSZE  \
0      1.0       2.0       2.0  5.397605e-79  5.397605e-79  1.000000e+00   
1      1.0       1.0       1.0  5.397605e-79  5.397605e-79  5.397605e-79   
2      1.0       2.0       2.0  5.397605e-79  5.397605e-79  2.000000e+00   
3      1.0       1.0       1.0  5.397605e-79  5.397605e-79  5.397605e-79   
4      1.0       5.0       5.0  5.397605e-79  2.000000e+00  5.397605e-79   

   DMDHRGND  DMDHRAGE  DMDHREDU  DMDHRMAR  INDFMIN2  
0       1.0      62.0       5.0       1.0      10.0  
1       1.0      53.0       3.0       3.0       4.0  
2       2.0      79.0       3.0       1.0       5.0  
3       2.0      56.0       5.0       6.0      10.0  
4       2.0      42.0       4.0       3.0       7.0

##### First: Change '7777's , '9999's, and  fill in Nans

In [304]:
# Fill DMQMILIZ, missing will be 2.0 (2.0 = No)
Demographics_df['DMQMILIZ'].fillna(2.0, inplace=True)

In [305]:
# Change all '99' to 1
Demographics_df.loc[Demographics_df['DMDBORN4'] == 99, 'DMDBORN4'] = 1

In [306]:
# Citizen Status
Demographics_df.loc[Demographics_df['DMDCITZN'] == 7, 'DMDCITZN'] = 2.0
Demographics_df.loc[Demographics_df['DMDCITZN'] == 9, 'DMDCITZN'] = 1.0
Demographics_df['DMDCITZN'].fillna(2.0, inplace=True)

In [307]:
# Head of Household's Education Level
Demographics_df.loc[Demographics_df['DMDHREDU'] == 9, 'DMDHREDU'] = 3.0
Demographics_df['DMDHREDU'].fillna(4.0, inplace = True)

In [308]:
# Head of Household's Marital Status
Demographics_df.loc[Demographics_df['DMDHRMAR'] == 77, 'DMDHRMAR'] = 5.0
Demographics_df.loc[Demographics_df['DMDHRMAR'] == 99, 'DMDHRMAR'] = 6.0
Demographics_df['DMDHRMAR'].fillna(1.0, inplace = True)

In [309]:
# Annual Family Income
Demographics_df.loc[Demographics_df['INDFMIN2'] == 77, 'INDFMIN2'] = 13.0
Demographics_df.loc[Demographics_df['INDFMIN2'] == 99, 'INDFMIN2'] = 12.0
Demographics_df['INDFMIN2'].fillna(12.0, inplace = True)

##### Now I need dummies
* Columns that need dummies:
    * RIAGENDR - Gender (1=Male, 2=Female)
        * Male = 1
        * Female = 0
    * RIDRETH3 - Race/Hispanic origin w/ NH Asian
        * Dummies (1,2,3,4,6,7)
    * DMQMILIZ - Served active duty in US Armed Forces
        * Yes = 1
        * No = 2
    * DMDBORN4 - Born in the US
        * Yes = 1
        * No = 2
    * DMDCITZN - US Citizenship status (1=Citizen, 2=Not a Citizen)
        * Yes = 1
        * No = 2
    * SIALANG - Language of Interview
        * English = 1
        * Spanish = 2
    * DMDHHSIZ - Total number of people in the Household
        * Dummies (1-7)
    * DMDFMSIZ - Total number of people in the Family
        * Dummies (1-7)
    * DMDHHSZA - Total number of children 5 years or younger in HH
        * Dummies (0-3)
    * DMDHHSZB - Total number of children 6-17 years old in HH
        * Dummies (0-4)
    * DMDHHSZE - Total number of adults 60 years or older in HH
        * Dummies (0-3)
    * DMDHRGND - Head of House ref persons gender
        * Male = 1
        * Female = 2
    * DMDHREDU - HH ref persons education level
        * Dummies (1-5)
    * DMDHRMAR - HH ref persons marital status
        * Dummies (1-6)
    * INDFMIN2 - Annual family income
        * Dummies (1-10, 12-15)

In [310]:
Demographics_df.shape

(9971, 18)

In [311]:
need_dummies = ['RIAGENDR',  'RIDRETH3', 'DMQMILIZ', 'DMDBORN4', 'DMDCITZN', 'SIALANG', 'DMDHHSIZ', 'DMDFMSIZ',
                'DMDHHSZA', 'DMDHHSZB', 'DMDHHSZE', 'DMDHRGND', 'DMDHREDU', 'DMDHRMAR', 'INDFMIN2']

In [312]:
Finished_Demographics_df = pd.get_dummies(Demographics_df, columns = need_dummies, drop_first = True)

In [313]:
Finished_Demographics_df.head()

SEQN  RIDAGEYR  DMDHRAGE  RIAGENDR_2.0  RIDRETH3_2.0  RIDRETH3_3.0  \
0  83732.0      62.0      62.0             0             0             1   
1  83733.0      53.0      53.0             0             0             1   
2  83734.0      78.0      79.0             0             0             1   
3  83735.0      56.0      56.0             1             0             1   
4  83736.0      42.0      42.0             1             0             0   

   RIDRETH3_4.0  RIDRETH3_6.0  RIDRETH3_7.0  DMQMILIZ_2.0      ...        \
0             0             0             0             1      ...         
1             0             0             0             1      ...         
2             0             0             0             0      ...         
3             0             0             0             1      ...         
4             1             0             0             1      ...         

   INDFMIN2_5.0  INDFMIN2_6.0  INDFMIN2_7.0  INDFMIN2_8.0  INDFMIN2_9.0  \
0             0             0             0             0             0   
1             0             0             0             0             0   
2             1             0             0             0             0   
3             0             0             0             0             0   
4             0             0             1             0             0   

   INDFMIN2_10.0  INDFMIN2_12.0  INDFMIN2_13.0  INDFMIN2_14.0  INDFMIN2_15.0  
0              1              0              0              0              0  
1              0              0              0              0              0  
2              0              0              0              0              0  
3              1              0              0              0              0  
4              0              0              0              0              0  

[5 rows x 58 columns]

In [318]:
Finished_Demographics_df = Finished_Demographics_df.rename(columns={'RIDAGEYR': 'Age', 'RIAGENDR_2.0': 'Sex_Female'})

In [319]:
Age_df = Finished_Demographics_df[['SEQN', 'Age']]

In [455]:
Age_df

SEQN   Age
0     83732.0  62.0
1     83733.0  53.0
2     83734.0  78.0
3     83735.0  56.0
4     83736.0  42.0
5     83737.0  72.0
6     83738.0  11.0
7     83739.0   4.0
8     83740.0   1.0
9     83741.0  22.0
10    83742.0  32.0
11    83743.0  18.0
12    83744.0  56.0
13    83745.0  15.0
14    83746.0   4.0
15    83747.0  46.0
16    83748.0   3.0
17    83749.0  17.0
18    83750.0  45.0
19    83751.0  16.0
20    83752.0  30.0
21    83753.0  15.0
22    83754.0  67.0
23    83755.0  67.0
24    83756.0  16.0
25    83757.0  57.0
26    83758.0  80.0
27    83759.0  19.0
28    83760.0   3.0
29    83761.0  24.0
...       ...   ...
9941  93673.0   5.0
9942  93674.0   8.0
9943  93675.0  38.0
9944  93676.0  35.0
9945  93677.0  34.0
9946  93678.0   1.0
9947  93679.0  72.0
9948  93680.0  17.0
9949  93681.0   5.0
9950  93682.0  41.0
9951  93683.0   3.0
9952  93684.0  34.0
9953  93685.0  53.0
9954  93686.0  13.0
9955  93687.0  16.0
9956  93688.0   1.0
9957  93689.0  69.0
9958  93690.0  32.0
9959  93691.0  25.0
9960  93692.0   1.0
9961  93693.0   7.0
9962  93694.0   7.0
9963  93695.0  76.0
9964  93696.0  26.0
9965  93697.0  80.0
9966  93698.0   2.0
9967  93699.0   6.0
9968  93700.0  35.0
9969  93701.0   8.0
9970  93702.0  24.0

[9971 rows x 2 columns]

In [456]:
Age_df = Age_df.drop(Age_df[Age_df.Age < 18].index)

In [458]:
Age_df.to_csv('Cleaned_Ages.csv')

In [358]:
Finished_Demographics_df = Finished_Demographics_df.drop(Finished_Demographics_df[Finished_Demographics_df.Age < 18].index)

In [422]:
Finished_Demographics_df.to_csv('Cleaned_Demographics.csv')

***

<div class="alert alert-block alert-info">

# Examination Data - Done

## Final DFs - 

#### Finished_Blood_Pressure_df  (5735, 6)

#### Body_Measures_df  (5735, 6)

</div>

### Next steps - Fill in NaN values

The NHANES datasets I'm using from the Examination section includes:
   * Blood Pressure
   * Body Measures

In [430]:
Preprocess_Blood_Pressure_df = pd.read_sas(filepath_or_buffer = 'Examination_Data_Blood_Pressure.XPT')
Preprocess_Body_Measures_df = pd.read_sas(filepath_or_buffer = 'Examination_Data_Body_Measures.XPT')

# Preprocess_Dentition_df = pd.read_sas(filepath_or_buffer = 'Examination_Data_Dentition.XPT')
# Preprocess_Dental_Recommends_df = pd.read_sas(filepath_or_buffer = 'Examination_Data_Oral_Recommends.XPT')

In [431]:
print (Preprocess_Blood_Pressure_df.shape)
print (Preprocess_Body_Measures_df.shape)

# print (Preprocess_Dentition_df.shape)
# print (Preprocess_Dental_Recommends_df.shape)

(9544, 21)
(9544, 26)


#### Blood Pressure Columns

   * BPXPLS - 60 sec. pulse (30 sec. pulse * 2)
        * Ages 8+
        * missing 2168 (probably mostly 0-7 year olds)
   * BPXPULS - Pulse regular or irregular?
        * missing 230
   * BPXSY2 - Systolic: Blood pres (2nd rdg) mm Hg
        * Ages 8+
   * BPXDI2 - Diastolic: Blood pres (2nd rdg) mm Hg
        * Ages 8+
    
    

    

In [432]:
blood_pressure_columns = ['SEQN', 'BPXPLS', 'BPXPULS', 'BPXSY2', 'BPXDI2']
Blood_Pressure_df = Preprocess_Blood_Pressure_df[blood_pressure_columns]

In [433]:
Blood_Pressure_df = Blood_Pressure_df.rename(columns={'BPXPLS': 'Heart Rate',
                                                      'BPXPULS': 'Irregular Pulse',
                                                      'BPXSY2': 'Systolic BP',
                                                      'BPXDI2': 'Diastolic BP'
                                                      })

#### Body Measure Columns
   * BMXWT - Weight (kg)
       * missing 99
   * BMXHT - Standing Height (cm)
       * Ages 2+
       * missing 775
       * may be able to fill this in from BMI and weight columns
   * BMXBMI - Body Mass Index (kg/m**2)
       * Ages 2+
       * missing 788

   * BMXWAIST - Waist Circumference (cm)
       * Ages 2+
       * Missing 1231
           
       
       
       
       
       
       
       

In [434]:
body_measure_columns = ['SEQN', 'BMXWT', 'BMXHT', 'BMXBMI', 'BMXWAIST']
Body_Measures_df = Preprocess_Body_Measures_df[body_measure_columns]

In [435]:
Body_Measures_df = Body_Measures_df.rename(columns={'BMXWT': 'Weight (kg)',
                                                      'BMXHT' : 'Standing Height (cm)',
                                                      'BMXBMI': 'BMI',
                                                      'BMXWAIST': 'Waist Circumference (cm)'})

In [436]:
Body_Measures_df.head()

SEQN  Weight (kg)  Standing Height (cm)   BMI  Waist Circumference (cm)
0  83732.0         94.8                 184.5  27.8                     101.1
1  83733.0         90.4                 171.4  30.8                     107.9
2  83734.0         83.4                 170.1  28.8                     116.5
3  83735.0        109.8                 160.9  42.4                     110.1
4  83736.0         55.2                 164.9  20.3                      80.4

# Cleaning Examination DF

##### Blood Pressure DF

   * BPXPLS - 60 sec. pulse (30 sec. pulse * 2)
        * Nan -> Age Median
   * BPXPULS - Pulse regular or irregular?
        * Reg = 1
        * Irr = 2
        * Nan -> 1 (mode = median)
   * BPXSY2 - Systolic: Blood pres (2nd rdg) mm Hg
        * Nan -> Age median
   * BPXDI2 - Diastolic: Blood pres (2nd rdg) mm Hg
        * Nan -> Age Median
    

In [437]:
# Add age column first, then drop everyone yonger than 18
Blood_Pressure_df = pd.merge(Blood_Pressure_df, Age_df, on="SEQN")
Blood_Pressure_df = Blood_Pressure_df.drop(Blood_Pressure_df[Blood_Pressure_df.Age < 18].index)
Blood_Pressure_df.head()

SEQN  Heart Rate  Irregular Pulse  Systolic BP  Diastolic BP   Age
0  83732.0        76.0              1.0        124.0          64.0  62.0
1  83733.0        72.0              1.0        140.0          88.0  53.0
2  83734.0        56.0              1.0        132.0          44.0  78.0
3  83735.0        78.0              1.0        134.0          68.0  56.0
4  83736.0        76.0              1.0        114.0          54.0  42.0

In [438]:
Blood_Pressure_Age_Medians = Blood_Pressure_df.groupby('Age').median()
Blood_Pressure_Age_Medians.head()

SEQN  Heart Rate  Irregular Pulse  Systolic BP  Diastolic BP
Age                                                                  
18.0  88073.0        74.0              1.0        112.0          62.0
19.0  89329.0        74.0              1.0        114.0          64.0
20.0  88944.0        74.0              1.0        114.0          66.0
21.0  88005.0        74.0              1.0        116.0          64.0
22.0  88480.0        72.0              1.0        112.0          62.0

In [439]:
# Fill in NaNs from the Median DF
Blood_Pressure_df.loc[Blood_Pressure_df['Heart Rate'].isnull(),'Heart Rate'] = Blood_Pressure_df['Age'].map(Blood_Pressure_Age_Medians['Heart Rate'])
Blood_Pressure_df.loc[Blood_Pressure_df['Irregular Pulse'].isnull(),'Irregular Pulse'] = Blood_Pressure_df['Age'].map(Blood_Pressure_Age_Medians['Irregular Pulse'])
Blood_Pressure_df.loc[Blood_Pressure_df['Systolic BP'].isnull(),'Systolic BP'] = Blood_Pressure_df['Age'].map(Blood_Pressure_Age_Medians['Systolic BP'])
Blood_Pressure_df.loc[Blood_Pressure_df['Diastolic BP'].isnull(),'Diastolic BP'] = Blood_Pressure_df['Age'].map(Blood_Pressure_Age_Medians['Diastolic BP'])



In [440]:
# Get Dummies for Blood Pressure

Finished_Blood_Pressure_df = pd.get_dummies(Blood_Pressure_df, columns = ['Irregular Pulse'], drop_first = True)


In [441]:
Finished_Blood_Pressure_df = Finished_Blood_Pressure_df.rename(columns={'Irregular Pulse_2.0': 'Irregular Pulse'})

In [423]:
Finished_Blood_Pressure_df.to_csv('Cleaned_Blood_Pressure.csv')

#### Body Measure Cleaning
   * BMXWT - Weight (kg)
       * missing 99
   * BMXHT - Standing Height (cm)
       * Ages 2+
       * missing 775
       * may be able to fill this in from BMI and weight columns
   * BMXBMI - Body Mass Index (kg/m**2)
       * Ages 2+
       * missing 788

   * BMXWAIST - Waist Circumference (cm)
       * Ages 2+
       * Missing 1231

In [442]:
# Add age column first, then drop everyone yonger than 18
Body_Measures_df = pd.merge(Body_Measures_df, Age_df, on="SEQN")
Body_Measures_df = Body_Measures_df.drop(Body_Measures_df[Body_Measures_df.Age < 18].index)
Body_Measures_df.head()

SEQN  Weight (kg)  Standing Height (cm)   BMI  Waist Circumference (cm)  \
0  83732.0         94.8                 184.5  27.8                     101.1   
1  83733.0         90.4                 171.4  30.8                     107.9   
2  83734.0         83.4                 170.1  28.8                     116.5   
3  83735.0        109.8                 160.9  42.4                     110.1   
4  83736.0         55.2                 164.9  20.3                      80.4   

    Age  
0  62.0  
1  53.0  
2  78.0  
3  56.0  
4  42.0

In [443]:
# Get Age Medians df
Body_Measures_Age_Medians = Body_Measures_df.groupby('Age').median()
Body_Measures_Age_Medians.head()

SEQN  Weight (kg)  Standing Height (cm)   BMI  \
Age                                                      
18.0  88073.0        70.20                167.80  24.2   
19.0  89329.0        70.50                168.40  24.1   
20.0  88944.0        72.90                168.90  25.6   
21.0  88005.0        76.20                166.90  26.3   
22.0  88480.0        74.65                168.35  26.0   

      Waist Circumference (cm)  
Age                             
18.0                     82.95  
19.0                     81.70  
20.0                     86.90  
21.0                     86.65  
22.0                     88.20

In [444]:
Body_Measures_df.loc[Body_Measures_df['Weight (kg)'].isnull(),'Weight (kg)'] = Body_Measures_df['Age'].map(Body_Measures_Age_Medians['Weight (kg)'])
Body_Measures_df.loc[Body_Measures_df['Standing Height (cm)'].isnull(),'Standing Height (cm)'] = Body_Measures_df['Age'].map(Body_Measures_Age_Medians['Standing Height (cm)'])
Body_Measures_df.loc[Body_Measures_df['Waist Circumference (cm)'].isnull(),'Waist Circumference (cm)'] = Body_Measures_df['Age'].map(Body_Measures_Age_Medians['Waist Circumference (cm)'])



In [445]:
Body_Measures_df

SEQN  Weight (kg)  Standing Height (cm)   BMI  \
0     83732.0         94.8                 184.5  27.8   
1     83733.0         90.4                 171.4  30.8   
2     83734.0         83.4                 170.1  28.8   
3     83735.0        109.8                 160.9  42.4   
4     83736.0         55.2                 164.9  20.3   
5     83737.0         64.4                 150.0  28.6   
9     83741.0         76.6                 165.4  28.0   
10    83742.0         64.5                 151.3  28.2   
11    83743.0         72.4                 166.1  26.2   
12    83744.0        108.3                 179.4  33.6   
15    83747.0         86.2                 176.7  27.6   
18    83750.0         76.2                 177.8  24.1   
20    83752.0         71.2                 163.6  26.6   
22    83754.0        117.8                 164.1  43.7   
23    83755.0         97.4                 183.8  28.8   
25    83757.0         80.5                 150.8  35.4   
26    83759.0        100.8                 175.4  32.8   
28    83761.0         61.8                 156.4  25.3   
29    83762.0        107.9                 168.5  38.0   
34    83767.0         59.0                 149.9  26.3   
36    83769.0         72.8                 170.7  25.0   
39    83773.0         67.7                 149.8  30.2   
41    83775.0         77.7                 160.2  30.3   
42    83776.0         56.6                 157.5  22.8   
43    83777.0         69.0                 166.1  25.0   
47    83781.0         87.8                 160.7  34.0   
49    83784.0         73.7                 170.7  25.3   
50    83785.0         75.6                 145.2  35.9   
51    83786.0        102.1                 182.2  30.8   
52    83787.0         77.4                 152.0  33.5   
...       ...          ...                   ...   ...   
9489  93645.0         70.3                 165.9   NaN   
9495  93652.0         88.9                 165.9  32.3   
9496  93653.0         76.9                 169.3  26.8   
9497  93654.0         54.5                 152.6  23.4   
9498  93655.0         75.9                 177.8  24.0   
9499  93656.0         92.7                 156.4  37.9   
9502  93659.0         67.9                 164.6  25.1   
9504  93661.0        101.8                 157.6  41.0   
9506  93663.0         75.7                 177.1  24.1   
9507  93664.0         74.9                 169.6  26.0   
9508  93665.0         69.0                 169.6  24.0   
9510  93668.0        119.6                 149.0  53.9   
9512  93670.0         81.4                 170.1  28.1   
9513  93671.0         89.0                 174.3  29.3   
9514  93672.0         81.9                 147.6  37.6   
9517  93675.0         77.6                 179.0  24.2   
9518  93676.0         92.2                 161.7  35.3   
9519  93677.0         62.2                 167.4  22.2   
9521  93679.0         57.8                 157.0  23.4   
9523  93682.0         58.2                 166.9  20.9   
9525  93684.0        101.2                 180.9  30.9   
9526  93685.0         78.7                 156.9  32.0   
9530  93689.0         64.8                 151.9  28.1   
9531  93690.0         89.5                 164.9  32.9   
9532  93691.0         39.2                 136.5  21.0   
9536  93695.0         59.1                 165.8  21.5   
9537  93696.0        112.1                 182.2  33.8   
9538  93697.0         71.7                 152.2  31.0   
9541  93700.0         78.2                 173.3  26.0   
9543  93702.0         58.3                 165.0  21.4   

      Waist Circumference (cm)   Age  
0                       101.10  62.0  
1                       107.90  53.0  
2                       116.50  78.0  
3                       110.10  56.0  
4                        80.40  42.0  
5                        92.90  72.0  
9                        86.60  22.0  
10                       93.30  32.0  
11                       82.95  18.0  
12                    

In [448]:
Body_Measures_df['BMI'] = Body_Measures_df['BMI'].fillna(Body_Measures_df['Weight (kg)'] / (Body_Measures_df['Standing Height (cm)'] / 100)**2)



In [449]:
Body_Measures_df

SEQN  Weight (kg)  Standing Height (cm)       BMI  \
0     83732.0         94.8                 184.5  27.80000   
1     83733.0         90.4                 171.4  30.80000   
2     83734.0         83.4                 170.1  28.80000   
3     83735.0        109.8                 160.9  42.40000   
4     83736.0         55.2                 164.9  20.30000   
5     83737.0         64.4                 150.0  28.60000   
9     83741.0         76.6                 165.4  28.00000   
10    83742.0         64.5                 151.3  28.20000   
11    83743.0         72.4                 166.1  26.20000   
12    83744.0        108.3                 179.4  33.60000   
15    83747.0         86.2                 176.7  27.60000   
18    83750.0         76.2                 177.8  24.10000   
20    83752.0         71.2                 163.6  26.60000   
22    83754.0        117.8                 164.1  43.70000   
23    83755.0         97.4                 183.8  28.80000   
25    83757.0         80.5                 150.8  35.40000   
26    83759.0        100.8                 175.4  32.80000   
28    83761.0         61.8                 156.4  25.30000   
29    83762.0        107.9                 168.5  38.00000   
34    83767.0         59.0                 149.9  26.30000   
36    83769.0         72.8                 170.7  25.00000   
39    83773.0         67.7                 149.8  30.20000   
41    83775.0         77.7                 160.2  30.30000   
42    83776.0         56.6                 157.5  22.80000   
43    83777.0         69.0                 166.1  25.00000   
47    83781.0         87.8                 160.7  34.00000   
49    83784.0         73.7                 170.7  25.30000   
50    83785.0         75.6                 145.2  35.90000   
51    83786.0        102.1                 182.2  30.80000   
52    83787.0         77.4                 152.0  33.50000   
...       ...          ...                   ...       ...   
9489  93645.0         70.3                 165.9  25.54245   
9495  93652.0         88.9                 165.9  32.30000   
9496  93653.0         76.9                 169.3  26.80000   
9497  93654.0         54.5                 152.6  23.40000   
9498  93655.0         75.9                 177.8  24.00000   
9499  93656.0         92.7                 156.4  37.90000   
9502  93659.0         67.9                 164.6  25.10000   
9504  93661.0        101.8                 157.6  41.00000   
9506  93663.0         75.7                 177.1  24.10000   
9507  93664.0         74.9                 169.6  26.00000   
9508  93665.0         69.0                 169.6  24.00000   
9510  93668.0        119.6                 149.0  53.90000   
9512  93670.0         81.4                 170.1  28.10000   
9513  93671.0         89.0                 174.3  29.30000   
9514  93672.0         81.9                 147.6  37.60000   
9517  93675.0         77.6                 179.0  24.20000   
9518  93676.0         92.2                 161.7  35.30000   
9519  93677.0         62.2                 167.4  22.20000   
9521  93679.0         57.8                 157.0  23.40000   
9523  93682.0         58.2                 166.9  20.90000   
9525  93684.0        101.2                 180.9  30.90000   
9526  93685.0         78.7                 156.9  32.00000   
9530  93689.0         64.8                 151.9  28.10000   
9531  93690.0         89.5                 164.9  32.90000   
9532  93691.0         39.2                 136.5  21.00000   
9536  93695.0         59.1                 165.8  21.50000   
9537  93696.0        112.1                 182.2  33.80000   
9538  93697.0         71.7                 152.2  31.00000   
9541  93700.0         78.2                 173.3  26.00000   
9543  93702.0         58.3                 165.0  21.40000   

      Waist Circumference (cm)   Age  
0                       101.10  62.0  
1                       107.90  53.0  
2                       116.50  78.0  
3       

In [450]:
Body_Measures_df.isnull().values.any()

False

In [451]:
Body_Measures_df.to_csv('Cleaned_Body_Measures.csv')

***